mypy/subtypes.py

```diff
@@ -249,13 +249,22 @@ def visit_overloaded(self, left: Overloaded) -> bool:
                     return True
             return False
         elif isinstance(right, Overloaded):
-            # TODO: this may be too restrictive
-            if len(left.items()) != len(right.items()):
+            if len(left.items()) < len(right.items()):
                 return False
-            for i in range(len(left.items())):
-                if not is_subtype(left.items()[i], right.items()[i], self.check_type_parameter,
+
+            # Ensure each overload in the left side is accounted for.
+            super_overloads = left.items()[:]
```

Comment:
@sixolet Bad variable name (I had originally written the code backwards by accident but forgot about this name), but not necessarily wrong.

I interpreted subtypes kind of like subsets: if X is a subset of Y, then every element in X is also an element in Y. The code here is basically checking that, if the left Overload is a subset of the right Overload, then every "element" in left should also be present in right (even though right can actually contain more elements).